hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -183,6 +185,14 @@
   public static final ConfigProperty<String> URL_ENCODE_PARTITIONING = KeyGeneratorOptions.URL_ENCODE_PARTITIONING;
   public static final ConfigProperty<String> HIVE_STYLE_PARTITIONING_ENABLE = KeyGeneratorOptions.HIVE_STYLE_PARTITIONING_ENABLE;
 
+  public static final List<String> PERSISTED_CONFIG_LIST = Arrays.asList(
+      Config.DATE_TIME_PARSER_PROP,
```

Comment:
does this mean, we are going to persist all key gen props to table config?
Also, do we also impose any constraints that one can't change the key gen props once the table is created?

Reply:
yes. These keygen props are immutable. 
And there are other configs that shouldn't be changed once the table is created. maybe we should open another ticket to track this, and manage them together.

Reply:
yes, please do open one.

Reply:
https://issues.apache.org/jira/browse/HUDI-3403